src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -487,16 +490,18 @@ private async Task ExecuteTestsInSourceAsync(IEnumerable<TestCase> tests, IRunCo
             // Queue the non parallel set
             if (nonParallelizableTestSet != null)
             {
+                _testRunCancellationToken?.ThrowIfCancellationRequested();
```

Comment:
nit: As you have the same line in the if/else, consider moving it once before the `if`